src/sinks/humio.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    sinks::splunk_hec::{Encoding, HecSinkConfig},
+    sinks::util::{BatchBytesConfig, TowerRequestConfig},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use serde::{Deserialize, Serialize};
+
+const HOST: &str = "https://cloud.humio.com";
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct HumioConfig {
+    token: String,
+    host: Option<String>,
+    encoding: Option<Encoding>,
+
+    #[serde(default)]
+    request: TowerRequestConfig,
+
+    #[serde(default)]
+    batch: BatchBytesConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<HumioConfig>("humio")
+}
+
+#[typetag::serde(name = "humio")]
+impl SinkConfig for HumioConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let host = self.host.clone().unwrap_or_else(|| HOST.to_string());
+        let encoding = self.encoding.clone().unwrap_or(Encoding::Json);
+
+        HecSinkConfig {
+            token: self.token.clone(),
+            host,
+            encoding,
+            batch: self.batch.clone(),
+            request: self.request.clone(),
+            ..Default::default()
+        }
+        .build(cx)
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
```

Comment:
We support metrics as well, how might we map that into Vector?

Reply:
I took another look through the docs but I may have missed a mention of metrics in the ingestion API. We could totally support metrics but I'm not aware of what shipping method you all prefer.

From what I know it probably doesn't make sense to ship metrics via the HEC endpoint? We could probably make another sink just for that.

Reply:
Let's put off the metrics ingest until we know log ingest is working well.